src/scripts/activate-user.ts

```diff
@@ -106,30 +106,77 @@ const dryRunActivateUser = (
   );
 
 /**
- * Activates the user.
- * @param service - the service to use to activate the user.
- * @param user - the user to activate.
- * @returns a TaskEither that resolves to the activated user.
+ * Prints out what would happen if we were to activate the user and send the activation email.
+ * @param user - the user to dry run the activation for.
+ * @returns a TaskEither that resolves to the user.
  */
-const activateUser = (
-  service: UserService,
+export const dryRunActivateUserAndSendEmail = (
   user: ActivatableUser
 ): TE.TaskEither<string, User> =>
   pipe(
     Console.info(
-      `Activating user [${user.id}] [${user.email}] with status [${user.status}]...`
+      `Will attempt to activate [${user.id}] [${user.email}] with status [${user.status}] and send activation email.`
     ),
     TE.rightIO,
     // eslint-disable-next-line functional/functional-parameters
-    TE.chain(() => service.activateUser(user.id)),
-    TE.tapIO((user) => Console.info(`Activated [${user.id}] [${user.email}].`)),
-    TE.chain((user) => validateUserExist(service, user.id)),
-    TE.tapIO((user) =>
+    TE.chain(() => TE.right(user))
+  );
+
+/**
+ * Activates the user.
+ * @param service - the service to use to activate the user.
+ * @param user - the user to activate.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+export const activateUser =
+  (service: UserService) =>
+  (user: ActivatableUser): TE.TaskEither<string, User> =>
+    pipe(
+      Console.info(
+        `Activating user [${user.id}] [${user.email}] with status [${user.status}]...`
+      ),
+      TE.rightIO,
+      // eslint-disable-next-line functional/functional-parameters
+      TE.chain(() => service.activateUser(user.id)),
+      TE.tapIO((user) =>
+        Console.info(`Activated [${user.id}] [${user.email}].`)
+      ),
+      TE.chain((user) => validateUserExist(service, user.id)),
+      TE.tapIO((user) =>
+        Console.info(
+          `User [${user.id}] [${user.email}] has new status [${user.status}].`
+        )
+      )
+    );
+
+/**
+ * Activates the user and sends the activation email.
+ * @param service - the service to use to activate the user.
+ * @param user - the user to activate.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+export const activateUserAndSendEmail =
```

Comment:
**Issue:** I don't like boolean flags, but I think having one function that takes a `sendEmail` parameter is better than this duplication.

If your function looks like this:
```
...
someFlag ? 
   doOneThing :
   doAnotherThing
```

it is a code smell and to be considered.

Reply:
modified in commit: 6b672568455dca1ecc7b7d13d462bde70475aa8b